packages/core/parcel-bundler/src/packagers/JSConcatPackager.js

```diff
@@ -283,17 +283,50 @@ class JSConcatPackager extends Packager {
       // so that they can be referenced by other modules directly.
       if (t.isVariableDeclaration(node)) {
         for (let decl of node.declarations) {
-          decls.push(t.variableDeclarator(decl.id));
-          if (decl.init) {
-            body.push(
-              t.expressionStatement(
-                t.assignmentExpression(
-                  '=',
-                  t.identifier(decl.id.name),
-                  decl.init
+          if (t.isObjectPattern(decl.id)) {
+            for (let prop of decl.id.properties) {
```

Comment:
I believe ObjectPattern and ArrayPattern work as the left hand side of an AssignmentExpression, in addition to a VariableDeclarator. e.g. `var {x} = y` -> `({x} = y)`. That would be smaller in the output.

Reply:
Done, and `getBindingIdentifiers` did indeed remove a whole if case